src/common/gui/CScalableBitmap.cpp

```diff
@@ -0,0 +1,63 @@
+#include "CScalableBitmap.h"
+#if MAC
+#include <CoreFoundation/CoreFoundation.h>
+#endif
+
+CScalableBitmap::CScalableBitmap( CResourceDescription desc ) : CBitmap( desc )
+{
+    int id = 0;
+    if( desc.type == CResourceDescription::kIntegerType )
+        id = (int32_t)desc.u.id;
+    
+    scales = {{ 100, 200 }};
+    scaleFilePostfixes[ 100 ] = "";
+    scaleFilePostfixes[ 200 ] = "@2x";
+
```

Comment:
Or?

```
    scales = {{100, 200}};
    scaleFilePostfixes[100] = "";
    scaleFilePostfixes[200] = "@2x";
```

Reply:
I would just have something like

```
int scaleFilePostfixes[2];
VSTGUI::CBitmap *scaleBitmaps[2];
```

The need to use `std::map` comes only from the fact that you have factored your scales.

Reply:
I don't have 2 sizes; I have 4 or 5 (1, 1.5, 2, 3, 4 at least). I chose an int because the 1.5 size is a float and I didn't want to hash float. When we do zoom properly this will be a longer list so I built it to scale to that.

Reply:
Per slack convo I will add a comment here explaining a bit more